plugins/unicodelib/unicodelib.cpp

```diff
@@ -286,28 +292,43 @@ class UPCList // User perceived character list
     uint32_t  length_;
     uint32_t  capacity_;
     bool invalid_;
-    void doCreateUPCList(BreakIterator& cbi) {
+    void doCreateUPCList(BreakIterator* cbi) {
         UErrorCode status = U_ZERO_ERROR;
         if (!capacity_) {
              capacity_ = ustring_.length();
          }
         next_ = new uint32_t[capacity_+1]; // the number of characters is always less or equal to the string length
         unsigned index=0;
-        cbi.setText(ustring_);
-        next_[index] = cbi.first();
-        for (int32_t end = cbi.next(); end != BreakIterator::DONE && length_ < capacity_; end = cbi.next())
+        cbi->setText(ustring_);
+        next_[index] = cbi->first();
+        for (int32_t end = cbi->next(); end != BreakIterator::DONE && length_ < capacity_; end = cbi->next())
         {
             length_++;
             next_[++index]=end;
         }
         if (U_FAILURE(status)) { length_ = 0; capacity_ = 0; invalid_ = true; }
     }
+    void doCreateUPCList() {
+        if (!capacity_) {
+             capacity_ = ustring_.length();
+         }
+        next_ = new uint32_t[capacity_+1]; // the number of characters is always less or equal to the string length
+        unsigned index=0;
+        next_[index] = 0;
+        int32_t end = 0;
+        while (end < capacity_)
+        {
+            end = end+ucpLength(ustring_[end]);
+            next_[++index] = end;
+            length_++;
+        }
+    }
 
 public:
-    UPCList(BreakIterator& cbi, const UnicodeString & source, uint32_t capacity=0)
+    UPCList(BreakIterator* cbi, const UnicodeString & source, uint32_t capacity=0)
         : length_(0), capacity_(capacity),ustring_(source), invalid_(false)
     {
-        doCreateUPCList(cbi);
+        (cbi == (const BreakIterator*)0)?doCreateUPCList():doCreateUPCList(cbi);
```

Comment:
or

  if (cbi)
    doCreateUPCList()
  else;
    doCreateUPCList(cbi);

Since they are functions that return void.